RotaryInvertedPendulum-arduino/PIDControl/PIDControl.ino

```diff
@@ -251,29 +334,24 @@ long convertDegreesToSteps(float degrees)
  */
 float convertRawAngleToDegrees()
 {
+    static long raw_prev = 0;
+    static bool first_reading = true;
+    static float position = 0.0f;
     // Get the current position of the AS5600
-    short ams5600_current_position = ams5600.getRawAngle();
-
-    // Calculate the difference between the current position and the initial position
-    short difference = ams5600_current_position - ams5600_initial_position;
-
-    if (difference < 0)
+    long raw = ams5600.getRawAngle();
+    if (first_reading) 
     {
-        difference += 4096;
+      raw_prev = raw;
+      first_reading = false;
     }
-
+    long delta = raw - raw_prev;
+    // Handle wrap around
+    if (delta > 2047) delta -= 4098;
+    if (delta < -2047) delta += 4098;
```

Comment:
Shouldn't it be `4096` instead of `4098`? I.e., `2^12` for the 12-bit AS5600 magnetic encoder?

Reply:
I am going to update the code as follows:
```suggestion
    if (delta > 2047) delta -= 4096;
    if (delta < -2047) delta += 4096;
```
Let me know if this is not correct, @VladimirIvan.